### PR TITLE
TestLabelsDemoApp: Replace isovalent/jobs-app by Opentelemetry demo app

### PR DESCRIPTION
**Pull request by Trung-DV**
Migrate e2e labels test from `isovalent/jobs-app` to Opentelemetry demo app
- Update the labels checker
- remove `-p 1 -parallel 1` flags when running tests

Fixes: #1976, #2176

### CODE REVIEW

**Review by mtardy — COMMENTED** [review]
Hey thanks a lot for doing this work, I think the patch works, failures are unrelated. Could you maybe squash your commits into one? 

I need to take a deeper look to see if we can make the label a bit more explicit maybe (add more of them if they are any) and test your patch locally in the meantime.

**Comment by mtardy** [comment]
> make the label a bit more explicit maybe (add more of them if they are any)

Those are the only ones set by the default helm chart install, so let's keep this it's plenty enough and it simplifies how they are checked!

File: tests/e2e/tests/labels/labels_test.go [hunk 1]
```diff
@@ -32,21 +33,13 @@ const (
...
-				helm.WithName("jobs-app"),
-				helm.WithChart("isovalent/jobs-app"),
-				helm.WithVersion("v0.7.1"),
-				helm.WithNamespace(namespace),
-				helm.WithArgs("--create-namespace", "--wait"),
+			if err := manager.RunUpgrade(
+				helm.WithArgs("onlineboutique"),
+				helm.WithArgs("oci://us-docker.pkg.dev/online-boutique-ci/charts/onlineboutique"),
+				helm.WithArgs("--install"),
+				helm.WithArgs("--create-namespace", "-n", namespace),
```
[the remark this answers is not in the document]
Comment (Trung-DV) [reply tests/e2e/tests/labels/labels_test.go:42]: Thank you @mtardy :D

**Comment by Trung-DV** [comment]
> > make the label a bit more explicit maybe (add more of them if they are any)
> 
> Those are the only ones set by the default helm chart install, so let's keep this it's plenty enough and it simplifies how they are checked!

As mentioned in https://github.com/cilium/tetragon/pull/2176#issuecomment-2063497524, I replaced labels to be checked by looked into https://github.com/GoogleCloudPlatform/microservices-demo/tree/90b81cca772eb61b8695a638e44b9b2cde70853f/helm-chart/templates and saw only the app label :D 

So just let it simple like this, right? @mtardy :D

File: tests/e2e/tests/labels/labels_test.go [hunk 2]
```diff
@@ -11,13 +11,14 @@ import (
...
+	"sigs.k8s.io/e2e-framework/pkg/features"
+	"sigs.k8s.io/e2e-framework/third_party/helm"
+
 	sm "github.com/cilium/tetragon/pkg/matchers/stringmatcher"
 	"github.com/cilium/tetragon/tests/e2e/checker"
 	"github.com/cilium/tetragon/tests/e2e/helpers"
 	"github.com/cilium/tetragon/tests/e2e/runners"
-	"sigs.k8s.io/e2e-framework/pkg/envconf"
-	"sigs.k8s.io/e2e-framework/pkg/features"
-	"sigs.k8s.io/e2e-framework/third_party/helm"
```
Comment (mtardy) [inline tests/e2e/tests/labels/labels_test.go:20]: and I'm sorry this is just a nit but could you remove this change? I don't think that's a linter that did that and it looks like we could avoid this little thing.

Comment (mtardy) [reply tests/e2e/tests/labels/labels_test.go:20]: And please suqashh the commits so that's there only one in the end :)

**Comment by mtardy** [comment]
> As mentioned in [#2176 (comment)](https://github.com/cilium/tetragon/pull/2176#issuecomment-2063497524), I replaced labels to be checked by looked into https://github.com/GoogleCloudPlatform/microservices-demo/tree/90b81cca772eb61b8695a638e44b9b2cde70853f/helm-chart/templates and saw only the app label :D
> 
> So just let it simple like this, right? @mtardy :D

This is perfect like this, thanks a lot!

File: tests/e2e/tests/labels/labels_test.go (hunk 2 above)
Comment (mtardy) [reply tests/e2e/tests/labels/labels_test.go:20]: aah okay I thought `goimports` was what most editors were using.

**Comment by mtardy** [comment]
So we have a failure on e2e tests but I think it's unrelated https://github.com/cilium/tetragon/actions/runs/8752369371/job/24019870334?pr=2345#step:7:1061. Since all the checkers have matched, I don't think it's related to your changes. Let me just investigate a bit before merging this.

**Comment by mtardy** [comment]
By the way, it seems the deployment time and checker time is a ~50% (or a bit more) gain from before, that's great for CI speed.

**Comment by Trung-DV** [comment]
> By the way, it seems the deployment time and checker time is a ~50% (or a bit more) gain from before, that's great for CI speed.

Relate to the CI speed, I have a concern why we need `-p 1 -parallel 1` flags here https://github.com/cilium/tetragon/blob/main/Makefile#L257? 
As I understand, `-p 1` means only 1 build command (for the test package) can run at a time, `-parallel 1` means only 1  execution of test functions that call `t.Parallel` running.

An with my experience, the `-parallel 1` seems redundant with `-p 1`. In case we want to make sure the tests do not run in parallel, we can set the `-parallel 1` and let the compiler build the test packages in parallel. That could reduce the waiting time when running testing (https://github.com/cilium/tetragon/actions/runs/8752369371/job/24024402667?pr=2345#step:7:10 to https://github.com/cilium/tetragon/actions/runs/8752369371/job/24024402667?pr=2345#step:7:12)

**Comment by mtardy** [comment]
> So we have a failure on e2e tests but I think it's unrelated https://github.com/cilium/tetragon/actions/runs/8752369371/job/24019870334?pr=2345#step:7:1061. Since all the checkers have matched, I don't think it's related to your changes. Let me just investigate a bit before merging this.

So in the end, even if the output is very confusing, it was related to this. Unfortunately the [microservices-demo](https://github.com/GoogleCloudPlatform/microservices-demo) does not ship arm64 images for all the microservices and it seems now it builds okay, they just don't want to publish the image for some reason https://github.com/GoogleCloudPlatform/microservices-demo/issues/622.

It worked fine locally on my computer since I use a rosetta share between my OS and the Linux VM I use and didn't realize the arch issue...

**Comment by mtardy** [comment]
So yeah again I'm sorry, you did a really good job and helped us a lot on something we have been postponing forever and you just bumped into this not being the ideal solution in the end.

At least now we know it, and we can see what actions should be taken:
- we can see if Google moves quickly on publishing the arm64 images;
- or if we should find an alternative real-life deployment;
- or just use something made-up that is simpler.

**Review by mtardy — CHANGES_REQUESTED** [review]
So yeah again I'm sorry, you did a really good job and helped us a lot on something we have been postponing forever and you just bumped into this not being the ideal solution in the end.

At least now we know it, and we can see what actions should be taken:
- we can see if Google moves quickly on publishing the arm64 images;
- or if we should find an alternative real-life deployment;
- or just use something made-up that is simpler.

Let me chat about that with the rest of the team and see what we can do! Thanks again for your PR :)

**Comment by Trung-DV** [comment]
Thank @mtardy for supporting me :D. While waiting for support from Google microservices demo, let me check that may we reduce the `go test` time 💪

**Comment by mtardy** [comment]
Hey, @lambdanis suggested I use this demo instead https://github.com/open-telemetry/opentelemetry-demo, apparently, it's a fork of the Google microservice demo and I tested it on arm64, and it works fine! You can find the doc here https://opentelemetry.io/docs/demo/kubernetes-deployment/.

If you still have the motivation to switch the e2e tests to use this one it would be amazing, otherwise, I can update the issue with the new desired deployment and someone can pick this up.

**Comment by Trung-DV** [comment]
> Hey, @lambdanis suggested we use this demo instead https://github.com/open-telemetry/opentelemetry-demo, apparently, it's a fork of the Google microservice demo and I tested it on arm64, and it works fine! You can find the doc here https://opentelemetry.io/docs/demo/kubernetes-deployment/.
> 
> 
> 
> If you still have the motivation to switch the e2e tests to use this one it would be amazing, otherwise, I can update the issue with the new desired deployment and someone can pick this up.

Thank you for your suggestion. Let me change the Helm charts, that would be a great start for me 💪

**Comment by Trung-DV** [comment]
Hi @mtardy , I have update the `labelsEventChecker` to match the otel-demo.
Since we have split e2e-tests to run separately, I suggest removing `-p 1 -parallel 1` so that the go's compiler can compile the test package in parallel (which is limited by -p 1)

**Review by jrfastab — APPROVED** [review]

**Comment by jrfastab** [comment]
LGTM we can follow up with the -p * if we need/want to.

**Comment by mtardy** [comment]
Lovely thanks!

**Pull request merged**